cmd/erasure-volume-meta.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+//go:generate msgp -file=$GOFILE -unexported
+type pool struct {
+	Name   string   `msg:"n"`
+	Local  string   `msg:"l"`
+	Remote []string `msg:"r"`
+}
+
+type volumeMeta struct {
+	Version int    `msg:"v"`
+	Token   string `msg:"t"`
+	Pools   []pool `msg:"ps"`
+}
+
+const volumeMetaFilename = minioMetaBucket + SlashSeparator + "volume.meta"
+
+func loadVolumeMeta(drives []string) (*volumeMeta, error) {
+	vm := &volumeMeta{}
+	for _, drive := range drives {
+		volumeMetaPath := pathJoin(drive, volumeMetaFilename)
+		buf, err := ioutil.ReadFile(volumeMetaPath)
+		if err != nil {
+			return nil, err
+		}
+		_, err = vm.UnmarshalMsg(buf)
+		return vm, err
+	}
+	return nil, errInvalidArgument
+}
+
+func saveVolumeMeta(drives []string) error {
+	vm := &volumeMeta{
+		Version: 1,
+		// Token: // TODO
+	}
+	poolRemoteHostnames := globalEndpoints.RemoteHostnames()
+	poolLocalHostnames := globalEndpoints.LocalHostnames()
+	pools := make([]pool, len(globalEndpoints))
+	for i := range globalEndpoints {
+		pools[i] = pool{
+			Local:  poolLocalHostnames[i],
+			Remote: poolRemoteHostnames[i],
+			Name:   "pool" + strconv.Itoa(i),
+		}
+	}
+	buf, err := vm.MarshalMsg(nil)
```

Comment:
Prepend a version for futureproofing:
```suggestion
	buf := make([]byte, 1, vm.Msgsize()+1)
	buf[0] = 1
	buf, err := vm.MarshalMsg(buf)
```
(check and remove when reading)

Reply:
Sure..